ra_aid/file_listing.py

```
@@ -0,0 +1,131 @@
+            raise DirectoryNotFoundError(f"Path is not a directory: {directory}")
+
+        result = subprocess.run(
+            ["git", "rev-parse", "--git-dir"],
+            cwd=str(path),
+            capture_output=True,
+            text=True
+        )
+        return result.returncode == 0
+
+    except subprocess.CalledProcessError as e:
+        raise GitCommandError(f"Git command failed: {e}")
```

Ruff (0.8.2)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)